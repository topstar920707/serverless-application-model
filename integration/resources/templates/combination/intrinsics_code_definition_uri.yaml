# Must support explicit bucket, key and version in CodeUri and DefinitionUri parameters

Parameters:
 Bucket:
  Type: String
 CodeKey:
  Type: String
 SwaggerKey:
  Type: String

Resources:

  MyLambdaFunction:
    Type: AWS::Serverless::Function
    Properties:
      Runtime: nodejs14.x
      Handler: index.handler
      MemorySize: 128
      CodeUri:
        Bucket:
          Ref: Bucket
        Key:
          Ref: CodeKey


  MyApi:
    Type: AWS::Serverless::Api
    Properties:
      StageName: FancyName
      DefinitionUri:
        Bucket:
          Ref: Bucket
        Key:
          Ref: SwaggerKey

