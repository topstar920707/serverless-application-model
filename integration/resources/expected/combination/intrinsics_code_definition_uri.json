[
  { "LogicalResourceId":"MyLambdaFunction", "ResourceType":"AWS::Lambda::Function" },
  { "LogicalResourceId":"MyLambdaFunctionRole", "ResourceType":"AWS::IAM::Role" },
  { "LogicalResourceId":"MyApiDeployment", "ResourceType":"AWS::ApiGateway::Deployment" },
  { "LogicalResourceId":"MyApi", "ResourceType":"AWS::ApiGateway::RestApi" },
  { "LogicalResourceId":"MyApiFancyNameStage", "ResourceType":"AWS::ApiGateway::Stage" }
]